File: gems/decomplex/examples/csharp/neglected-updates-initializers.cs
public class TaskConfig {
    public int stack_size { get; set; }
    public int timeout { get; set; }
}

public class ServerConfig {
    public int port { get; set; }
    public int timeout { get; set; }
}

public class Test {
    public static void setup_task1() {
        var c = new TaskConfig { stack_size = 1024, timeout = 10 };
    }

    public static void setup_task2() {
        var c = new TaskConfig { stack_size = 2048, timeout = 20 };
    }

    public static void setup_task3() {
        var c = new TaskConfig { stack_size = 4096, timeout = 30 };
    }

    public static void setup_server() {
        var c = new ServerConfig { port = 8080, timeout = 30 };
    }

    public static void partial_setup() {
        // Missing timeout, should be flagged as neglected-update on TaskConfig.
        // Without normalization, it gets conflated with ServerConfig's literal initialization.
        var c = new TaskConfig { stack_size = 1024 };
    }
}

File: gems/decomplex/examples/zig/neglected-updates-initializers.zig
const TaskConfig = struct {
    stack_size: u32,
    timeout: u32,
};

const ServerConfig = struct {
    port: u32,
    timeout: u32,
};

pub fn setup_task1() void {
    const c = TaskConfig{ .stack_size = 1024, .timeout = 10 };
    _ = c;
}

pub fn setup_task2() void {
    const c = TaskConfig{ .stack_size = 2048, .timeout = 20 };
    _ = c;
}

pub fn setup_task3() void {
    const c = TaskConfig{ .stack_size = 4096, .timeout = 30 };
    _ = c;
}

pub fn setup_server() void {
    const c = ServerConfig{ .port = 8080, .timeout = 30 };
    _ = c;
}

pub fn partial_setup() void {
    // Missing timeout, should be flagged as neglected-update on TaskConfig.
    // Without normalization, it gets conflated with ServerConfig's literal initialization.
    const c = TaskConfig{ .stack_size = 1024 };
    _ = c;
}
